### PR TITLE
feat(ui): Adds ability to go to project's issue stream from globalHeader

### LINKED ISSUE #16538: Navigate to Project Issues from Issue Detail view
## Summary

Enable users to consistently navigate from issue detail view to list of issues in the same project.
Possible implementation is to make the project name in the header be a hyperlink to the list of project issues.

## Motivation

> Why should this be worked on? What problems or use cases does it solve or
improve?

Consider this flow:
1. Go to https://sentry.io/issues/
2. Click an issue belonging to "project A"
3. Clicking the <- arrow navigates back to all issues, no way to navigate to all issues of "project A" except for using the browser's Back button

image

## Additional Context

Related issue: #13864.
I noticed that the behavior of the Back button is hard to predict and is influenced by actions you do in the issue view. Going "back" can be ambiguous -- back to where I came from? Back to all issues? Back to project issues? Therefore what I propose is to click on the project name to navigate to project issues, or an equivalent solution.

### PR DESCRIPTION
**Pull request by matejminar**
Fixes #16538

### CODE REVIEW

File: src/sentry/static/sentry/app/components/organizations/multipleProjectSelector.jsx [hunk 1]
```diff
@@ -133,6 +135,32 @@ export default class MultipleProjectSelector extends React.PureComponent {
     this.setState({hasChanges: true});
   };
 
+  renderProjectName() {
+    const {location} = this.context.router;
+    const {forceProject, multi, organization} = this.props;
+
+    if (forceProject && multi) {
+      return (
+        <Tooltip title={t('Project Issues Stream')} position="bottom">
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/organizations/multipleProjectSelector.jsx]: Does this wording make sense?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/organizations/multipleProjectSelector.jsx]: I think just `Issues Stream` is good

**Review by billyvg — APPROVED** [review]

**Comment by BYK** [comment]
@matejminar is this also related #16776?

**Comment by matejminar** [comment]
The back button takes you to the issue stream with your `location.search`. 
https://github.com/getsentry/sentry/blob/c96ad8a01203d81547d7d37a5aa16ba580574476/src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/index.jsx#L500-L513
So if you go to event detail from issue stream and you got two projects filtered at that time, the back button should take you to the issue stream with that two projects filtered. (because of `GlobalSelectionLink`)

Thanks to this PR, on event detail page, you can click on the project name
image
to consistently navigate to issue stream of that one project.

The issue you mentioned is interesting, because coming from email, there should be no project specified in URL and filled there only after loading the site. Therefore the back button should take you to the filtered issue stream of the project id we fill into the url upon loading the event detail page.

**Pull request merged**